### PR TITLE
Authorization problem

### PR DESCRIPTION
**Pull request by sanjivani1812**
Some function parameter issues

### CODE REVIEW

**Comment by coolbung** [comment]
What parameter issues are you facing ?

**Comment by sanjivani1812** [comment]
client id is not passed to check function and loadActionByCode function client and action are typecast.

File: src/administrator/libraries/action.php [hunk 1]
```diff
@@ -188,7 +188,7 @@ public function bind(&$array)
 	 *
 	 * @since __DEPLOY_VERSION__
 	 */
-	public static function loadActionByCode(String $code, String $client)
+	public static function loadActionByCode($code, $client)
```
Comment (coolbung) [inline src/administrator/libraries/action.php]: Why did you remove the typecasting ? Do you see any integer only values of code or client ?

File: src/administrator/includes/rbacl.php [hunk 2]
```diff
@@ -154,10 +154,12 @@ public static function check($userId, $client, $action, $contentId = null)
 	 */
 	public static function authorise($userId, $client, $action, $contentId = null)
 	{
+		$client = (string) $client;
+		$action = (string) $action;
 		$user = Factory::getUser($userId);
 		$result = $user->authorise($action, $client);
 
-		return $result && self::check($userId, $client, $action);
+		return $result && self::check($userId, $client, $action, $contentId);
```
Comment (coolbung) [inline src/administrator/includes/rbacl.php]: @thite-amol you need to look at this.

Comment (sanjivani1812) [reply src/administrator/includes/rbacl.php]: @thite-amol  - Please check this comments
